### ISSUE TITLE
Is the implementation of ITG in blip2_qformer correct?

### ISSUE THREAD

**Issue by zhangyuygss** [salesforce/LAVIS#436]
I ran the code of blip2 pretrian, and got size mismatch error of attention mask and input.
It leads to this [line of code](https://github.com/salesforce/LAVIS/blob/main/lavis/models/blip2_models/blip2_qformer.py#L260).
To my understanding of the paper, the input here should be the query embeds, but the input `decoder_input_ids` is actully the text tokens. 
The `attention_mask` is the concat of text_attns and query_attns, which has different length with the input of qforer.
Is their something wrong with the ITG implementation? Or did I misunderstand the code?
